### ISSUE TITLE
-Wmissing-braces in drivers/gpu/drm/amd/display

### ISSUE THREAD

**Issue by nathanchance** [ClangBuiltLinux/linux#345]
```
drivers/gpu/drm/amd/amdgpu/../display/amdgpu_dm/amdgpu_dm.c:2314:38: warning: suggest braces around initialization of subobject [-Wmissing-braces]
        struct dc_surface_dcc_cap output = {0};
                                            ^
                                            {}
drivers/gpu/drm/amd/amdgpu/../display/amdgpu_dm/amdgpu_dm.c:2314:38: warning: suggest braces around initialization of subobject [-Wmissing-braces]
        struct dc_surface_dcc_cap output = {0};
                                            ^
                                            {}
drivers/gpu/drm/amd/amdgpu/../display/amdgpu_dm/amdgpu_dm.c:5089:60: warning: suggest braces around initialization of subobject [-Wmissing-braces]
                struct dc_surface_update dummy_updates[MAX_SURFACES] = { 0 };
                                                                         ^ 
                                                                         {}
3 warnings generated.

drivers/gpu/drm/amd/amdgpu/../display/dc/core/dc_link_dp.c:50:57: warning: suggest braces around initialization of subobject [-Wmissing-braces]
        union training_aux_rd_interval training_rd_interval = {0};
                                                               ^
                                                               {}
1 warning generated.
```

Patches sent (I opted to keep them separate for the "Fixes" tag, maybe I should have put them together):

https://lore.kernel.org/lkml/20190201201226.12812-1-natechancellor@gmail.com/
https://lore.kernel.org/lkml/20190201201428.13357-1-natechancellor@gmail.com/
https://lore.kernel.org/lkml/20190201201541.13680-1-natechancellor@gmail.com/

**Comment by nathanchance** [comment]
All three patches accepted:

https://cgit.freedesktop.org/~agd5f/linux/commit/?h=amd-staging-drm-next&id=1816cb976526a85e317137e08e921d06c6ed800f

https://cgit.freedesktop.org/~agd5f/linux/commit/?h=amd-staging-drm-next&id=a0c15edcb46be65770cf8c374bfce21991d78190

https://cgit.freedesktop.org/~agd5f/linux/commit/?h=amd-staging-drm-next&id=e1be4cb583800db36ed7f6303f7a8c205be24ceb

**Comment by nathanchance** [comment]
Fixed in `next-20190212`:

https://git.kernel.org/next/linux-next/c/8daa12182a7ab441a2b3eeb170f9d8a12b206d6c

https://git.kernel.org/next/linux-next/c/19afd79951e6303dbe0d15293a2f66ceb4c772f6

https://git.kernel.org/next/linux-next/c/d6d36b55a0f394a3895349285ed6980278809489
